pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -173,6 +169,122 @@ func (configgen *ConfigGeneratorImpl) buildClusters(proxy *model.Proxy, req *mod
 	return resources, model.XdsLogDetails{AdditionalInfo: fmt.Sprintf("cached:%v/%v", cacheStats.hits, cacheStats.hits+cacheStats.miss)}
 }
 
+func (configgen *ConfigGeneratorImpl) deltaServices(updatedService model.ConfigKey, proxy *model.Proxy, push *model.PushContext,
+	watched []string) ([]*model.Service, []string) {
+	deletedClusters := make([]string, 0)
+	services := make([]*model.Service, 0)
+	service := push.ServicesForHostname(proxy, host.Name(updatedService.Name), updatedService.Namespace)
+	// push.ServiceForHostname will return nil if the proxy doesn't care about the service OR it was deleted.
+	// we can cross-reference with WatchedResources to figure out which services were deleted.
+	if service == nil {
```

Comment:
`ServicesForHostname` returns a list. This means this will ~never be `nil`, and we end up inserting nil entries and panic later on.